File: android/DKParkingSDK/src/main/kotlin/com/dkparking/sdk/backend/BackendConfig.kt
// BackendConfig.kt
// DK Parking Engine SDK — Android Version 1
// Single place to configure the Render backend URL and API key.
// Set this once at app startup before calling TelemetryUploader or DatasetClient.

package com.dkparking.sdk.backend

/**
 * Backend configuration for the DK Parking Engine.
 * Set [BackendConfig.shared] at app startup before any SDK calls.
 */
data class BackendConfig(
    /** Base URL of the DK Parking backend on Render. e.g. https://dk-parking-backend.onrender.com */
    val baseUrl: String,
    /** API key sent in every request as X-API-Key header. Must match MOBILE_API_KEY on Render. */
    val apiKey: String
) {
    companion object {
        /** Shared instance — set once at app startup. */
        var shared: BackendConfig = BackendConfig(
            baseUrl = "https://dk-parking-backend.onrender.com",
            apiKey  = ""   // Set via BackendConfig.shared = BackendConfig(...) at startup
        )
    }
}

File: ios/DKParkingSDK/Sources/DKParkingSDK/Backend/BackendConfig.swift
// BackendConfig.swift
// DK Parking Engine SDK — iOS Version 1
// Single place to configure the Render backend URL and API key.
// Set these before calling TelemetryUploader or DatasetClient.

import Foundation

public struct BackendConfig {
    /// Base URL of the DK Parking backend on Render.
    /// Example: https://dk-parking-backend.onrender.com
    public let baseURL: URL

    /// API key sent in every request as X-API-Key header.
    /// Must match MOBILE_API_KEY set on Render environment.
    public let apiKey: String

    public init(baseURL: URL, apiKey: String) {
        self.baseURL = baseURL
        self.apiKey = apiKey
    }

    /// Shared instance — set this once at app startup before any SDK calls.
    public static var shared: BackendConfig = BackendConfig(
        baseURL: URL(string: "https://dk-parking-agent.onrender.com")!,
        apiKey: ""   // Set via BackendConfig.shared = BackendConfig(baseURL:apiKey:) at startup
    )
}
